core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -433,14 +433,17 @@ object TopicCommand extends Logging {
   /**
     * ensures topic existence and throws exception if topic doesn't exist
     *
-    * @param opts
-    * @param topics
-    * @param topicOptWithExists
+    * @param foundTopics Topics that were found to match the requested topic name.
+    * @param requestedTopic Name of the topic that was requested.
+    * @param requireTopicExists Indicates if the topic needs to exist for the operation to be successful.
+    *                           If set to true, the command will throw an exception if the topic with the
+    *                           requested name does not exist.
     */
-  private def ensureTopicExists(topics: Seq[String], topicOptWithExists: Boolean = false) = {
-    if (topics.isEmpty && !topicOptWithExists) {
+  private def ensureTopicExists(foundTopics: Seq[String], requestedTopic: Option[String], requireTopicExists: Boolean = true) = {
+    // If no topic name was mentioned, do not need to throw exception.
+    if (requestedTopic.isDefined && requireTopicExists && foundTopics.isEmpty) {
       // If given topic doesn't exist then throw exception
-      throw new IllegalArgumentException(s"Topics in [${topics.mkString(",")}] does not exist")
+      throw new IllegalArgumentException(s"Topics in [${foundTopics.mkString(",")}] does not exist")
```

Comment:
This message seems not very clear. We know `foundTopics` must be empty, so the message would look like this:
> Topics in [] does not exist

I think it makes more sense for the message to indicate that the requested topic was not found. For example:

> Topic 'foo' does not exist as expected

Reply:
I know about this issue and had initially fixed this in this pull request. But I later realized that there is a separate JIRA https://issues.apache.org/jira/browse/KAFKA-8053 for it, and I can push a different commit for it.

I am not aware of the general practice used for Apache Kafka, but assumed that we try to address separate bugs as separate commits, to improve traceability.

I totally intend to follow this commit up with another PR to fix just the string for the exception. Let me know if you disagree and want me to club the fixes for the two JIRAs together.

Reply:
Yeah, that's a good point. The two fixes largely overlap. So we can either fix both issues here or we can leave the fixed message for #6381. @scholzj Do you have any preference?

Reply:
@hachikuji I don't have a strong preference. If you think its easier to fix it here in this PR and close mine I'm fine with it.

Reply:
Okay, @hachikuji I have made the string change here. We can resolve both JIRAs once this is merged.